In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')

In [4]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7662 non-null   object 
 1   Year             7662 non-null   int64  
 2   Value            7662 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 239.6+ KB


5. How many rows and columns are in `gdp_df`? What are the data types of each column?
-ANSWER: 4 columns and 7662 rows; The data types are float, integer and object.

In [ ]:
gdp_df.shape

6. Drop the `Value Footnotes` column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [5]:
gdp_df = gdp_df.drop(columns = ['Value Footnotes'])
gdp_df.columns = ['Country', 'Year', 'GDP_Per_ Capita']
gdp_df.head()

Country  Year  GDP_Per_ Capita
0  Afghanistan  2021      1517.016266
1  Afghanistan  2020      1968.341002
2  Afghanistan  2019      2079.921861
3  Afghanistan  2018      2060.698973
4  Afghanistan  2017      2096.093111

7. How many countries have data for all years? Which countries are missing many years of data? Look at the number of observations per year. What do you notice? 